src/main/kotlin/io/snyk/plugin/ui/settings/ScanTypesPanel.kt

```diff
@@ -219,25 +221,31 @@
             return
         }
 
+        var sastSettingsError: String = ""
+        val sastCliConfigSettings: CliConfigSettings? =
+        try {
+            getSnykApiService().getSastSettings()
+        } catch (t: ClientException) {
+            val defaultErrorMsg = "Your org's SAST settings are misconfigured."
+            val userMessage = if (t.message.isNullOrEmpty()) defaultErrorMsg else t.message!!
+            SnykBalloonNotificationHelper.showError(userMessage, null)
+            sastSettingsError = userMessage;
+            null
+        }
+
+        settings.sastOnServerEnabled = sastCliConfigSettings?.sastEnabled
+        settings.localCodeEngineEnabled = sastCliConfigSettings?.localCodeEngine?.enabled
+        settings.localCodeEngineUrl = sastCliConfigSettings?.localCodeEngine?.url
         val snykCodeAvailable = isSnykCodeAvailable(settings.customEndpointUrl)
         showSnykCodeAlert(
-            if (snykCodeAvailable) "" else "Snyk Code only works in SAAS mode (i.e. no Custom Endpoint usage)"
+            if (sastSettingsError.isNotEmpty()) sastSettingsError else ""
```

Comment:
I get a warning in IntelliJ here - please use `ifEmpty`

Reply:
Done